### PR TITLE
94 wall

### PR DESCRIPTION
**Pull request by RubyKobayashi**
- [ ] 1. User is redirected to their wall when they log in 
- [x] 2. /:user_id 
- [x] 3. User can go to any /:user_id to see each user’s wall
- [x] 4. User can’t see any other user’s wall without logging in
- [x] 5. User is redirected to login page
- [x] 6. User can see other users’ wall when logged in
- [ all posts appear in feed as well as on the specified wall] 7. User can post on any user’s wall and the post only appears on that user's wall
- [yes for delete, yet to be implemented for update ] 8. User can not update or delete posts or comments apart from their own (on any wall)

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -6,17 +6,23 @@ def new
...
 
   def update
-    @post.update_attributes(message: post_params)
-    redirect_to posts_url
+    @post.update_attributes(message: params[:post][:message])
+    redirect_to "/#{@post.wall}"
+  end
+
+  def wall
+    no_such_wall unless User.where(id: params[:wall]).count == 1
```
Comment (ch359) [inline app/controllers/posts_controller.rb:23]: how about
`no_such_wall unless User.find_by_id(params[:wall])` as something a little simpler? Might save a DB query if nothing else.

**Review by Estevenson1994 — APPROVED** [review]
splendid

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<%= link_to new_post_path, id: 'new_post' do %>
+<%= link_to "#{current_user.id}/posts/new", id: 'new_post' do %>
```
Comment (ch359) [inline app/views/posts/index.html.erb:1]: I'd be interested to understand why not new_post_path?

File: app/views/posts/new.html.erb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <%= form_for @post do |form| %>
   <%= form.label :message %>
   <%= form.text_field :message %>
+  <%= form.hidden_field :wall, :value => params[:wall] %>
```
Comment (ch359) [inline app/views/posts/new.html.erb:4]: Nice!

**Review by ch359 — APPROVED** [review]
One possible tweak to consider...

File: app/views/posts/index.html.erb (hunk 2 above)
Comment (david-mears) [reply app/views/posts/index.html.erb:1]: new_post_path goes to /posts/new, but we need a way to remember which wall we are going to post to.
Methods in order:
show_wall
new
create
- #new needs to remember the wall id from show_wall to pass it to create.

**Pull request merged**